app/services/setting_service.rb

```diff
@@ -0,0 +1,70 @@
+require_relative "data_sources/json_setting_data_source"
+require_relative "../shared/logging_module"
+require_relative "../shared/available_settings"
+require_relative "./services"
+
+module FastlaneCI
+  class SettingServiceKeyNotFoundError < FastlaneCIError
+  end
+
+  # Provides access to system settings
+  class SettingService
+    include FastlaneCI::Logging
+    attr_accessor :setting_data_source
+
+    def initialize(setting_data_source: nil)
+      unless setting_data_source.nil?
+        unless setting_data_source.class <= SettingDataSource
+          raise "setting_data_source must be descendant of #{SettingDataSource.name}"
+        end
+      end
+
+      if setting_data_source.nil?
+        # Default to JSONSettingDataSource
+        # TODO: do we need `sample_data` here?
+        logger.debug(
+          "setting_data_source is new, using `ENV[\"data_store_folder\"]` " \
+          "if available, or `sample_data` folder"
+        )
+        data_store_folder = ENV["data_store_folder"] # you can set it at runtime!
```

Comment:
We should revisit this separately. `data_store_folder` is used in multiple places and might not be what we want.